tests/integration/targets/ec2/tasks/main.yml

```diff
@@ -0,0 +1,141 @@
+---
+- module_defaults:
+    group/aws:
+      aws_region: '{{ aws_region }}'
+      aws_access_key: '{{ aws_access_key | default(omit) }}'
+      aws_secret_key: '{{ aws_secret_key | default(omit) }}'
+      security_token: '{{ security_token | default(omit) }}'
+  block:
+
+    # SETUP: vpc, ec2 key pair, subnet, security group, ec2 instance
+    - name: create a VPC to work in
+      ec2_vpc_net:
+        cidr_block: '{{ vpc_cidr }}'
+        state: present
+        name: '{{ resource_prefix }}_setup'
+        resource_tags:
+          Name: '{{ resource_prefix }}_setup'
+      register: setup_vpc
+
+    - name: create a key pair to use for creating an ec2 instance
+      ec2_key:
+        name: '{{ resource_prefix }}_setup'
+        state: present
+      register: setup_key
+
+    - name: create a subnet to use for creating an ec2 instance
+      ec2_vpc_subnet:
+        az: '{{ ec2_region }}a'
```

Comment:
I would recommend against setting the AZ this way.  In this example we don't care which AZ it lands in, and for a handful or regions you're not guaranteed that Availability Zone 'a' will exist.

Reply:
@gravesm  You can see an example of picking an available AZ for a region in these tests: https://github.com/ansible-collections/community.aws/blob/main/tests/integration/targets/ec2_instance/roles/ec2_instance/tasks/env_setup.yml#L3-L13